ballerina/client.bal

```diff
@@ -256,6 +256,61 @@ public isolated client class Client {
         http:Response httpResponse = check self.amazonS3->delete(requestURI, request);
         return handleHttpResponse(httpResponse);
     }
+    
+    # Generates a presigned URL for the object.
+    #
+    # + bucketName - The name of the bucket  
+    # + objectName - The name of the object  
+    # + expirationTime - The time until the presigned URL is valid in seconds
+    # + httpMethod - The HTTP method to be used, either GET or PUT  
+    # + partNo - The part number of the object, when uploading multipart objects
+    # + uploadId - The upload ID of the multipart upload
+    # + return - If success, a presigned URL, else an error
+    @display {label: "Create Presigned URL"}
+    remote isolated function createPresignedURL(
+            @display {label: "Bucket Name"} string bucketName,
+            @display {label: "Object Name"} string objectName,
+            @display {label: "HTTP Method"} string httpMethod,
```

Comment:
Since only GET and PUT are valid, we can define two constants and use the union of that here

```ballerina
const HTTP_METHOD_GET = "GET";
const HTTP_METHOD_PUT = "PUT";
```

```suggestion
            @display {label: "HTTP Method"} HTTP_METHOD_GET|HTTP_METHOD_PUT httpMethod,
```